### PR TITLE
fixes #7592 - adding typed to mockgen

### LINKED ISSUE #7592: Use typed mocks
### Area for Improvement

We can update our use of gomock to generate typed functions by passing in `-typed`. 

From the command line help:

```
  -typed
    	Generate Type-safe 'Return', 'Do', 'DoAndReturn' function
```

### Observed behavior

We don't pass in `-typed` so none of the gomock generated code is strongly typed. This slows down development for us, because it's possible to get the wrong signatures.

### Desired behavior

Use `-typed` everywhere we create a mock. This shouldn't break us if we already wrote correct code.

### Proposed Fix

- Do a search for `//go:generate mockgen`. Replace with `//go generate mockgen -typed`.
- Run `make generate-go`.
- Run the tests (should work). 

### rad Version

asdf

### Operating system

_No response_

### Additional context

_No response_

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#12367](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/12367)

### PR DESCRIPTION
**Pull request by superbeeny**
# Description

Fixes issue #7592 - adding types to generated mocks

## Type of change

- This pull request fixes a bug in Radius and has an approved issue #7592 .

Fixes: #7592

### CODE REVIEW

**Review by lakshmimsft — DISMISSED** [review]

File: pkg/cli/clients/management_mocks.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ import (
 // Because these interfaces are non-exported, they MUST be defined in their own file
 // and we MUST use -source on mockgen to generate mocks for them.
 
-//go:generate mockgen -typed -source=./management_mocks.go -destination=./mock_management_wrapped_clients.go -package=clients -self_package github.com/radius-project/radius/pkg/cli/clients github.com/radius-project/radius/pkg/cli/clients genericResourceClient,applicationResourceClient,environmentResourceClient,resourceGroupClient
+//go:generate mockgen -typed -typed -source=./management_mocks.go -destination=./mock_management_wrapped_clients.go -package=clients -self_package github.com/radius-project/radius/pkg/cli/clients github.com/radius-project/radius/pkg/cli/clients genericResourceClient,applicationResourceClient,environmentResourceClient,resourceGroupClient
```
Comment (kachawla) [inline pkg/cli/clients/management_mocks.go:37]: Looks like this one was already using typed..

```suggestion
//go:generate mockgen -typed -source=./management_mocks.go -destination=./mock_management_wrapped_clients.go -package=clients -self_package github.com/radius-project/radius/pkg/cli/clients github.com/radius-project/radius/pkg/cli/clients genericResourceClient,applicationResourceClient,environmentResourceClient,resourceGroupClient
```

**Review by kachawla — COMMENTED** [review]
Thanks for doing this @superbeeny!

**Comment by ytimocin** [comment]
Thanks for your contribution @superbeeny! You should run `make generate` in your local and then commit the generated changes.

**Comment by ytimocin** [comment]
One last thing :) DCO is failing: https://github.com/radius-project/radius/pull/7647/checks?check_run_id=25736485013. You should sign your commit. I guess, if you do `git commit -s --amend` and push all the changes again, it should be signed. Thanks @superbeeny!

**Comment by superbeeny** [comment]
There are days I really hate git 👎

File: pkg/ucp/queue/client/client_test.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func TestStartDequeuer(t *testing.T) {
 		Data:        []byte("{}"),
 	}, nil)
 
-	secondCall := mockCli.EXPECT().Dequeue(gomock.Any(), gomock.Any()).Return(nil, ErrInvalidMessage).After(firstCall)
+	secondCall := mockCli.EXPECT().Dequeue(gomock.Any(), gomock.Any()).Return(nil, ErrInvalidMessage).After(firstCall.Call)
```
Comment (ytimocin) [inline pkg/ucp/queue/client/client_test.go:53]: This is one of the changes that come with the mock updates, I guess.

Comment (ytimocin) [reply pkg/ucp/queue/client/client_test.go:53]: I think we are using this (second call should be after the first call test) only in this test.

**Review by ytimocin — APPROVED** [review]
Thanks for this change @superbeeny! 🚀

**Review by kachawla — APPROVED** [review]

**Review by lakshmimsft — APPROVED** [review]

**Comment by lakshmimsft** [comment]
Thanks @superbeeny. I'm merging this in.

**Pull request merged**